apps/rewards/app/components/App/App.js

```diff
@@ -190,8 +191,9 @@ class App extends React.Component {
       startBlock -= reward.duration
     }
     if(reward.rewardType === ONE_TIME_DIVIDEND){
-      startBlock = currentBlock
-      reward.duration = millisecondsToBlocks(Date.now(), reward.dateReference)
+      const rawBlockDuration = millisecondsToBlocks(Date.now(), reward.dateReference)
+      startBlock = reward.dateReference <= new Date() ? currentBlock + rawBlockDuration - BLOCK_PADDING : currentBlock 
+      reward.duration = reward.dateReference <= new Date() ? BLOCK_PADDING : rawBlockDuration
```

Comment:
@chadoh how ugly is this? This feels a bit like bike-shedding but should I just switch to an `if-else` control structure?

Reply:
I think an `if/else` would be easier to follow!